### ISSUE TITLE
DateTimePicker onSubmit control

### ISSUE THREAD

**Issue by nitin-1926** [mantinedev/mantine#4579]
### What package has an issue

@mantine/dates

### Describe the bug

When we try to override the submitButtonProps like this:
```
submitButtonProps={{
    onClick: event => {
        // Do something
},
}}
```
It overrides the onClick of the ActionIcon and thus doesn't close the picker.
This is happening because submitButtonProps is spread above the onClick thus it get's overridden
![Screenshot 2023-07-27 at 3 00 49 PM](https://github.com/mantinedev/mantine/assets/44253685/3df67ded-d341-4212-90d7-66419c7fc93f)

### What version of @mantine/hooks page do you have in package.json?

6.0.1

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

There are two possible ways to handle this.

1. Expose the control to directly open and close the picker component.
2. Spread the submitButtonProps after the "onClick" props in "ActionIcon"

**Comment by nitin-1926** [comment]
Although there is a workaround for it. We can control the popover props directly by penetration and manually open and close the modal like this.

But it took me sometime to figure it out. Posting it here in case someone else is looking for it.

<img width="1102" alt="Screenshot 2023-07-27 at 3 06 34 PM" src="https://github.com/mantinedev/mantine/assets/44253685/6b4dced6-5f24-4d97-8fdc-9a1815bfec10">

**Comment by rtivital** [comment]
`submitButtonProps.onClick` works correctly – it is called when the button is clicked. On your screenshot this logic is on line 227. Sandbox that demonstrates behavior – https://codesandbox.io/s/old-fog-lrlhx3?file=/src/App.tsx

**Comment by nitin-1926** [comment]
@rtivital When I tried this earlier, it was not working. I agree the logic on line number 227 is correct. But my point was that the submitButtonProps was getting spread before the onClick prop thus if I am sending "onClick" in "submitButtonProps", it was overriding and therefore the modal was not getting closed.
On a side note, the sandbox that you have shared works. Is the reason that when I raised this, I was using 6.0.1 and in the sandbox 6.0.19 version is used?
